modules/javafx.graphics/src/test/java/test/javafx/scene/layout/HeaderBarTest.java

```diff
@@ -0,0 +1,438 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.layout;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.geometry.Dimension2D;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.layout.HeaderBar;
+import javafx.scene.shape.Rectangle;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import test.util.ReflectionUtils;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class HeaderBarTest {
+
+    HeaderBar headerBar;
+
+    @BeforeEach
+    void setup() {
+        headerBar = new HeaderBar();
+    }
+
+    @Test
+    void emptyHeaderBar() {
+        assertNull(headerBar.getLeading());
+        assertNull(headerBar.getCenter());
+        assertNull(headerBar.getTrailing());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+        "TOP_LEFT, 10, 10, 100, 80",
+        "TOP_CENTER, 10, 10, 100, 80",
+        "TOP_RIGHT, 10, 10, 100, 80",
+        "CENTER_LEFT, 10, 10, 100, 80",
+        "CENTER, 10, 10, 100, 80",
+        "CENTER_RIGHT, 10, 10, 100, 80",
+        "BOTTOM_LEFT, 10, 10, 100, 80",
+        "BOTTOM_CENTER, 10, 10, 100, 80",
+        "BOTTOM_RIGHT, 10, 10, 100, 80"
+    })
```

Comment:
these tests can be simplified to avoid parsing with @CsvSource and default conversion, example:

```
    @ParameterizedTest
    @CsvSource(value = {"yo,true,1", 
                        "bah,true,1.1", 
                        "meh,false,2.2"})
    void test(String candidate, boolean expected, double a) {
        System.out.println("candidate=" + candidate + " expected=" + expected + " a=" +a );
    }
```